CorePluginFramework/test/unit/src/au/gov/asd/tac/constellation/plugins/parameters/types/BooleanParameterTypeNGTest.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2010-2021 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.parameters.types;
+
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameter;
+import au.gov.asd.tac.constellation.plugins.parameters.types.BooleanParameterType.BooleanParameterValue;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author aldebaran30701
+ */
+public class BooleanParameterTypeNGTest {
+
+    public BooleanParameterTypeNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of build method, of class BooleanParameterType.
+     */
+    @Test
+    public void testBuild_String() {
+        System.out.println("build_string");
+        String id = "booleanParameter";
```

Comment:
You could reuse the `BooleanParameterType.ID` instead but did you pick a different string on purpose to show that it should still work?

Reply:
I just passed any string in to ensure it was set correctly in the constructor. Could be a good idea to use the .ID but I would also want to test this to ensure it wasn't hardcoding anything. (Choosing a random string like "booleanParameter" makes it unlikely)